### DISCUSSION TITLE
How to use a library when building an extension?

### DISCUSSION THREAD

**Discussion by reply2future** [FreshRSS/Extensions#155] [Q&A]
I want to use a library like [this](https://github.com/orhanerday/open-ai) in my extension, but I have no idea how to achieve it because there is no composer.json. By the way, I'm a beginner in PHP.

**Comment by Alkarex** [comment]
Did you succeed ( Cf. https://github.com/FreshRSS/Extensions/pull/153 , https://github.com/reply2future/xExtension-NewsAssistant ) or is it still necessary?
You can ship libraries along with your own code. You can use composer to download whatever is needed, but all the needed resources must be available in your repository.

**Reply by reply2future** [reply]
I used the pure code to complete the task. Thanks for helping.
